### PR TITLE
Add `HTTP::Cookie#expire`

### PR DESCRIPTION
**Pull request by a-alhusaini**
# Reasoning

I tried to use `cookies.delete("...")` today and it didn't cause the cookie to expire from the browser. It sent it again on the next request.

This method sets the cookie value to an empty string and seets the expiry date to 5 minutes ago. Browsers automatically delete expired cookies.

My aim is to provide a convenience to anyone who happens to need to use `HTTP::Cookie`

### CODE REVIEW

**Comment by ysbaddaden** [comment]
Since the intent is to **expire** the cookie, then `#expire` would be a better name than `#destroy`.

I'd also put an expiration date far further than 5 minutes ago (clocks can get unsynced)and use a fixed date, like the UNIX timestamp (1970-01-01 00:00:00 UTC).

**Comment by a-alhusaini** [comment]
@ysbaddaden Hey julian! This little function is a convenience for destroying cookies (which usually means setting their expiry to some date in the past and setting their value to blank)

I just thought others might want a convenient function to destroy cookies. 

Although as you pointed out, it is superflous. I just PR'd it to promote developer happiness :3

**Comment by straight-shoota** [comment]
It would be much appreciated to follow [the guidelines for feature discussions](https://github.com/crystal-lang/crystal/blob/master/.github/PULL_REQUEST_TEMPLATE/pull_request_template.md) and open an issue for a new proposal first, to keep the general discussion independent of a PR with a concrete implementation.
Since this is relatively trivial, I suppose we can continue the discussion here. But next time, please open an issue first please.

**Comment by straight-shoota** [comment]
Actually, setting `Expires` to a date in the past does not fully ensure the cookie to expire. [`Max-Age`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#max-agenumber) takes precedence. If `Max-Age` is set, the value of `Expires` doesn't matter.

So I think the functionality should be implemented as `self.max_age = Time::Span.zero`. 
We could also consider `self.expires = Time::UNIX_EPOCH` on top, just for good measure, but that should not really be necessary.

**Comment by a-alhusaini** [comment]
Sorry about the typo. I can't run the test suite locally so I'm relying on the CI.

File: src/http/cookie.cr [hunk 1]
```diff
@@ -192,6 +192,16 @@ module HTTP
...
 
+    # tell the browser to delete the cookie.
+    # Set its value to an empty string
+    # and set its expiration time to the past
+    # Browsers delete cookies that have expired.
+    def destroy
+      self.value = ""
+      self.expires = Time::UNIX_EPOCH
+      self.max_age = Time::Span.zero
+    end
```
Comment (Sija) [inline src/http/cookie.cr:203]:
```suggestion
    # Tells the browser to expire the cookie by setting its 
    # max age time to zero and its expiration time to the past.
    def expire
      self.expires = Time::UNIX_EPOCH
      self.max_age = Time::Span.zero
    end
```

Comment (a-alhusaini) [reply src/http/cookie.cr:203]: Could you elaborate on why you suggest that we remove setting the value to a blank string?

Comment (Sija) [reply src/http/cookie.cr:203]: Sure, my reasoning was twofold:

1. `expire` method name IMO implies only cookie expiration, not emptying its value
2. Spec doesn't mention (or I've not seen it does) clearing out the value as the requirement for a cookie expiration

Comment (a-alhusaini) [reply src/http/cookie.cr:203]: I believe destroy will better represent what the function does. And there are many methods with a name similar to `#expire` I think calling it `#destroy` would be more clear.

Comment (straight-shoota) [reply src/http/cookie.cr:203]: Clearing the value is not a requirement for expiration. But I think it makes a lot of sense *not* to send the value you want gone to the client again.

File: src/http/cookie.cr [hunk 2]
```diff
@@ -192,6 +192,16 @@ module HTTP
       end
     end
 
+    # tell the browser to delete the cookie.
+    # Set its value to an empty string
+    # and set its expiration time to the past
+    # Browsers delete cookies that have expired.
```
Comment (straight-shoota) [inline src/http/cookie.cr:198]: **suggestion:** Improve wording.

```suggestion
    # Destroys the cookie.
    # This is done by causing the cookie to expire. Also clears its value.
```

**Review by straight-shoota — APPROVED** [review]

File: src/http/cookie.cr [hunk 3]
```diff
@@ -192,6 +192,14 @@ module HTTP
       end
     end
 
+    # Destroys the cookie.
+    # This is done by causing the cookie to expire. Also clears its value.
+    def destroy
+      self.value = ""
```
Comment (Sija) [inline src/http/cookie.cr:198]: I still do think that's a bit unexpected to clear the value in here. Just sayin'

**Review by bcardiff — COMMENTED** [review]
I think that #expire or #expire! would reflect better the behavior.

There was a issue in the past for rails where the behavior of #delete for cookies was discussed. An outcome of that discussion was that can be applied here is that HTTP::Cookies type is not only used for browsers. For example #destroy in the context of a MITM proxy would have a different expectation. So I think #expire or #expire! would be better and more application agnostic.

**Comment by ysbaddaden** [comment]
I initialized proposed `#expire` so I wholeheartedly agree with @bcardiff.

**Review by ysbaddaden — APPROVED** [review]

**Pull request merged**